### PR TITLE
Used tax payer number and country validation in Core library.

### PR DESCRIPTION
**Pull request by abdallahbeshi**

### CODE REVIEW

File: src/Mews.Fiscalization.Spain.Tests/Basics.cs [hunk 1]
```diff
@@ -21,23 +23,23 @@ public class Basics
 
         public static readonly LocalCompany IssuingCompany = new LocalCompany(
             new LimitedString120(System.Environment.GetEnvironmentVariable("issuer_name") ?? "INSERT_ISSUER_NAME"),
-            new TaxPayerNumber(System.Environment.GetEnvironmentVariable("issuer_tax_number") ?? "INSERT_ISSUER_TAX_NUMBER")
+            new TaxpayerIdentificationNumber(new Country("ES"), System.Environment.GetEnvironmentVariable("issuer_tax_number") ?? "INSERT_ISSUER_TAX_NUMBER")
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Spain.Tests/Basics.cs:26]: European country and european taxPayer number?

File: src/Mews.Fiscalization.Spain/Model/Request/RequestModel.cs [hunk 2]
```diff
@@ -259,22 +259,22 @@ public static bool IsValid(LimitedString1to20 id, LimitedString120 name, Country
...
+                return Core.Model.Country.IsValid(countryCode);
             }
             return id != null && name != null;
         }
     }
 
     public class InvoiceId
     {
-        public InvoiceId(TaxPayerNumber issuer, LimitedString1to60 number, DateTime date)
+        public InvoiceId(TaxpayerIdentificationNumber issuer, LimitedString1to60 number, DateTime date)
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Spain/Model/Request/RequestModel.cs:270]: Let's also split this to class per file. (also response)

**Review by KaliCZ — APPROVED** [review]

File: src/Mews.Fiscalization.Spain/Model/Request/CounterPartyCompany.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public sealed class CounterPartyCompany : Coproduct2<LocalCompany, ForeignCompany>
+    {
+        public CounterPartyCompany(LocalCompany companyTitle)
+            : base(companyTitle)
+        {
+            if (companyTitle == null)
+            {
+                throw new ArgumentNullException(nameof(companyTitle));
+            }
```
Comment (marektresnak) [inline src/Mews.Fiscalization.Spain/Model/Request/CounterPartyCompany.cs:14]: We could replace these with `Check` from Core.

**Review by marektresnak — APPROVED** [review]

File: src/Mews.Fiscalization.Spain/Model/Request/CounterPartyCompany.cs (hunk 3 above)
Comment (abdallahbeshi) [reply src/Mews.Fiscalization.Spain/Model/Request/CounterPartyCompany.cs:14]: We have a YT for that, there are many usages that we need to change.

**Pull request merged**